api/environments/secret_handler.go

```diff
@@ -58,6 +58,16 @@ func (eh EnvironmentHandler) ChangeEnvironmentComponentSecret(appName, envName,
 		secretObjName = strings.TrimSuffix(secretName, defaults.BlobFuseCredsAccountNamePartSuffix)
 		partName = defaults.BlobFuseCredsAccountNamePart
 
+	} else if strings.HasSuffix(secretName, defaults.CsiAzureCredsAccountKeyPartSuffix) {
```

Comment:
The CsiAzureCredsAccountKeyPartSuffix suffix is the same as the BlobFuseCredsAccountKeyPartSuffix suffix, so this branch in the if...elseif block will never be called I think. Instead the elseif for BlobFuseCredsAccountKeyPartSuffix will be called. The correct secret will still be updated.

We have so many different secret types (normal secrets, dns, client certificates, misc volume mounts( now, so we should probably think about categorizing them in the api and web console, and send the category as part of the request when setting secrets.

Reply:
Agree, we'll need to extend meta-info to properly compare. Current logic works properly only because content  is exactly the same.